extensions/amp-ad/0.1/test/test-amp-ad-3p-impl.js

```diff
@@ -212,14 +208,21 @@ function tests(name) {
       })).to.be.not.be.rejected;
     });
 
+    it('should only layout once', () => {
+      return getAd().then(ad => {
+        const firstLayout = ad.implementation_.layoutCallback();
+        ad.implementation_.unlayoutCallback();
+        const newLayout = ad.implementation_.layoutCallback();
+        ad.implementation_.unlayoutCallback();
+        const secondLayout = ad.implementation_.layoutCallback();
+        expect(firstLayout).to.not.equal(newLayout);
+        expect(newLayout).to.equal(secondLayout);
```

Comment:
Because there's a runtime layout, that's the new layout. `firstLayout`  is the duplicate layout of runtime layout. `newLayout` is the layout after I can `unlayoutCallback` and is a real new layout. Maybe rename `firstLayout` to `runtimeLayoutDupl`?